### PR TITLE
[Paddle Inference] add constant folding pass

### PR DESCRIPTION
**Pull request by zhoutianzi666**
### PR types

Others
### PR changes

Others
### Describe

 When a op's inputs and outputs is determined before feeding data to the model, we can remove this op from the model. 
This ConstantFolding pass can remove all these like ops.

| 模型                                     | 关闭    | 启用pass |
| ---------------------------------------- | ------- | -------- |
| SwinTransformer_base_patch4_window12_384 | 45.4006 | 44.8208  |
| hifigan_csmsc_static_0.1.1               | 25.2497 | 25.016   |

### CODE REVIEW

File: paddle/fluid/framework/ir/constant_folding_pass.cc [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+};
+}  // namespace patterns
+
+ConstantFoldingPass::ConstantFoldingPass() {}
+
+void ConstantFoldingPass::ApplyImpl(ir::Graph *graph) const {
+  PADDLE_ENFORCE_NOT_NULL(
+      graph, platform::errors::PreconditionNotMet("graph should not be null."));
+  FusePassBase::Init("constant_folding", graph);
+  auto *scope = param_scope();
```
Comment (zhangjun) [inline paddle/fluid/framework/ir/constant_folding_pass.cc:60]: 增加nulltpr判断

File: paddle/fluid/framework/ir/constant_folding_pass.cc [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    for (auto in_node : op_node->inputs) {
+      map[in_node->Name()] = 0;
+      if (!in_node->Var()->Persistable()) {
+        input_persis = false;
+      }
+    }
+    for (auto out_node : op_node->outputs) {
+      map[out_node->Name()] = 0;
+    }
+    // Forbide other node in graph having the same name with nodes in map
```
Comment (zhangjun) [inline paddle/fluid/framework/ir/constant_folding_pass.cc]: forbid

**Review by zhangjun — APPROVED** [review]
LGTM

**Review by cyj1986 — APPROVED** [review]
LGTM

**Pull request merged**